### ISSUE TITLE
Reproduce Multilingual BERT results

### ISSUE THREAD

**Issue by chitwansaharia** [google-research/bert#996]
I am trying to reproduce Multilingual Bert (BASE) results as given here. https://github.com/google-research/bert/blob/master/multilingual.md

I don't find information regarding some hyper-parameters like batch_size, number of training steps, learning rate etc. I guess these might be different from monolingual BERT (whose information is present in the paper) since it uses a vocabulary size 3-4 times larger than the monolingual version.

**Comment by dhruvsakalley** [comment]
This is exactly what I'm interested in as well, I'd like to understand what was the # tokens/ text volume it was trained on and how much data from each language contributed? What was the train time/ cycles so that this can work can be extended. If there is an arxiv around this, that would be awesome to include in the readme. Thanks!

**Comment by chitwansaharia** [comment]
I was able to reproduce (upto 1% margin) Multi Lingual BERT results (on XNLI zero-shot evaluation) using the same configuration as monolingual BERT. Although I only train with 128 sequences (and a batch size of 1024) for 1M steps. I don't finetune it using 512 length sequences for 100k steps as they do in monolingual setting. Doing that might further improve the performance.

**Comment by stefan-it** [comment]
Hi @chitwansaharia , just a question regarding to pre-processig: did you perform sentence-splitting for all of the languages :thinking:  I'm not aware of any library that could handle all languages (spacy supports only a few, but you could use NLTK to train unsupervised models and then perform sentence splitting).

Did you perfom up/down sampling for specific languages? Would be highly interested in these things, although this issue here is older > 1 year 😅

**Comment by jerryIsHere** [comment]
> Hi @chitwansaharia , just a question regarding to pre-processig: did you perform sentence-splitting for all of the languages 🤔 I'm not aware of any library that could handle all languages (spacy supports only a few, but you could use NLTK to train unsupervised models and then perform sentence splitting).
> 
> Did you perfom up/down sampling for specific languages? Would be highly interested in these things, although this issue here is older > 1 year 😅

I am facing same problem
In multilingual.md, they said "For all other languages, we apply the same recipe as English: (a) lower casing+accent removal, (b) punctuation splitting, (c) whitespace tokenization. ".

In https://github.com/google-research/bert#tokenization, it said "Punctuation splitting: Split all punctuation characters on both sides (i.e., add whitespace around all punctuation characters). Punctuation characters are defined as (a) Anything with a P* Unicode class, (b) any non-letter/number/space ASCII character (e.g., characters like $ which are technically not punctuation). E.g., john johanson's, → john johanson ' s ,".

I wonder is this how they to next sentence prediction? 🤔 Perhaps even if it isn't, it is still good enough.
